util/cli/cli.go

```diff
@@ -29,6 +30,42 @@ import (
 	"github.com/argoproj/argo-cd/util/io"
 )
 
+type LogWriter struct{}
+
+var levelRegex *regexp.Regexp
+
+const (
+	LevelError   = "error"
+	LevelWarning = "warning"
+	LevelFatal   = "fatal"
+	LevelPanic   = "panic"
+)
+
+func init() {
+	var err error
+	levelRegex, err = regexp.Compile("level=([a-z]+)")
+	if err != nil {
+		log.WithError(err).Fatal("Cannot setup log level")
+	}
+}
+
+func (w *LogWriter) detectLogLevel(p []byte) (level string) {
+	matches := levelRegex.FindStringSubmatch(string(p))
+	if len(matches) > 1 {
+		level = matches[1]
+	}
+	return
+}
+
+func (w *LogWriter) Write(p []byte) (n int, err error) {
+	level := w.detectLogLevel(p)
+
+	if level == LevelError || level == LevelWarning || level == LevelFatal || level == LevelPanic {
```

Comment:
I don't see profit of "Level*" variables since there is no way to customize logic of Writer

what do you think of this implementation?
```
levelRegex, err = regexp.Compile("level=(panic|fatal|warning|error)")
// ...
if levelRegex.Match(p) {
  return os.Stderr.Write(p)
}
return os.Stderr.Write(p)
```

Reply:
Makes sense 🙂